pmd-core/src/main/java/net/sourceforge/pmd/lang/rule/properties/PropertyDescriptorUtil.java

```diff
@@ -88,4 +89,53 @@ public static String typeIdFor(Class<?> valueType, boolean multiValue) {
         return null;
     }
 
+
+    /**
+     * Gets a builder for a required fields map.
+     *
+     * @return A builder
+     *
+     * @see ExpectedFieldsBuilder
+     */
+    public static ExpectedFieldsBuilder expectedFields() {
+        return new ExpectedFieldsBuilder();
+    }
+
+
+    /** Builder for an expected fields map. */
+    public static final class ExpectedFieldsBuilder {
+
+        Map<PropertyDescriptorField, Boolean> requiredFields = new HashMap<>();
```

Comment:
`private final`

I use this as the default for all fields when I'm writing a new class.  I widen scope, and allow mutability when needed.